test/prioritized-pools.test.js

```diff
@@ -0,0 +1,261 @@
+import * as sql from '../src';
+import serialWarmupStrategy from '../src/serial-warmup-strategy';
+import delay from './delay';
+
+let connection;
+const realSetTimeout = setTimeout;
+jest.useFakeTimers();
+
+describe('prioritized pools tests', () => {
+  afterEach(() => {
+    connection.close();
+  });
+  it(`lower priority pool starts out as the primary but is replaced by a higher priority
+    pool after a prioritize cycle completes`, () => {
+    connection = new sql.ConnectionPoolParty({
+      // the dsns are the same, but are sufficient for these tests
+      dsns: [
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 1,
+        },
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 0,
+        },
+      ],
+      prioritizePools: true,
+      // this helps prevent warmup from resolving before all the pools are created
+      warmupStrategy: serialWarmupStrategy,
+    });
+    return connection.warmup()
+      .then(() => {
+        // confirm priority1 pool is first and priority0 is second
+        expect(connection.pools[0].dsn.priority).toEqual(1);
+        expect(connection.pools[1].dsn.priority).toEqual(0);
+        // confirm both pools are connected
+        expect(connection.pools[0].connection.connected).toEqual(true);
+        expect(connection.pools[1].connection.connected).toEqual(true);
+        // force a prioritize cycle
+        jest.runOnlyPendingTimers();
+      })
+      .then(() => {
+        // confirm priority1 pool is now second and priority0 is now first
+        expect(connection.pools[0].dsn.priority).toEqual(0);
+        expect(connection.pools[1].dsn.priority).toEqual(1);
+      });
+  });
+  it(`lower priority pool starts out as the primary. higher priority pool is unhealthy.
+    the higher priority pool is healed and placed as primary after a prioritize cycle completes.`, () => {
+    connection = new sql.ConnectionPoolParty({
+      // the dsns are the same, but are sufficient for these tests
+      dsns: [
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 1,
+        },
+        {
+          user: 'sa',
+          password: 'PoolPartyyy9000',
+          server: 'localhost',
+          database: 'PoolParty',
+          priority: 0,
+        },
+      ],
+      prioritizePools: true,
+      // this helps prevent warmup from resolving before all the pools are created
+      warmupStrategy: serialWarmupStrategy,
+    });
+    return connection.warmup()
+      .then(() => {
```

Comment:
probably you can make your tests a bit easier to read by adding async-await

Reply:
agreed, i'm planning on doing a complete refactor of this library for async/await.  i wanted to keep everything consistent until that point though.